streams/src/main/java/org/apache/kafka/streams/state/StreamsMetadata.java

```diff
@@ -87,24 +117,29 @@ public boolean equals(final Object o) {
         if (!stateStoreNames.equals(that.stateStoreNames)) {
             return false;
         }
-        return topicPartitions.equals(that.topicPartitions);
+        if (!topicPartitions.equals(that.topicPartitions)) {
+            return false;
+        }
+        if (!standbyStateStoreNames.equals(that.standbyStateStoreNames)) {
+            return false;
+        }
 
+        return standbyTopicPartitions.equals(that.standbyTopicPartitions);
```

Comment:
Do you mean replace calling `a.equals(b)`, with `Objects.equals(a, b)`? It will give us null checking. so yes. Will change.. 
I don't follow "custom" adjective.. We have to compare each field right anyway right?

Reply:
and FWIW its how it is on master as well https://github.com/apache/kafka/blob/39158bd944f7c64e32634cec8171515b93466ffa/streams/src/main/java/org/apache/kafka/streams/state/StreamsMetadata.java#L83 

Will make a sweep and change them